### DISCUSSION TITLE
How to use the persist plugin with reusable components?

### DISCUSSION THREAD

**Discussion by jadersbr** [alpinejs/alpine#2580] [1. Help]
I'm building a datatable and I would like to persist the filters. For that I started using the "persist" component. The problem I'm facing is that the key for persist (field) is the same in all datatables. 

For example, the **openFilters** key is the same in the users datatable and the permissions datatable, as I am reusing the component.

Well, I thought about using the `.as` modifier and concatenating it with the datatable id, example: users-data-table-**openFilters**, permissions-data-table-**openFilters**, but it didn't work.

```
document.addEventListener('alpine:init', () => {
        Alpine.data('dataTable', function () {
            return {
                dataTableId: '',
                allCheckBoxIsChecked: false,
                eventCreateForm: '',
                directionColumn: '',
                withTrashed: this.$persist(false).as(this.dataTableId + '-withTrashed').using(sessionStorage),
                openFilters: this.$persist(false).as(this.dataTableId + '-openFilters').using(sessionStorage),
                archivable: false,
```

The problem is that the dataTableId is not available at alpine init(), so I always get undifinied.

```
init() {
       this.dataTableId = this.data.parameters.dataTableId;
```

![image](https://user-images.githubusercontent.com/58928002/149604318-b02adca2-4cb0-49e6-bf29-24d2e4bda741.png)

Any tips to fix this problem?

**Comment by SimoTod** [comment]
You can pass a string and use that one to generate a unique alias.
See https://alpinejs.dev/globals/alpine-data#initial-parameters
